app/models/validations/shared_validations.rb

```diff
@@ -79,9 +79,9 @@ def location_during_startdate_validation(record)
 
     if location_inactive_status.present?
```

Comment:
I think this validation wants to live in a different file to this one - for that same reason that you've split out where the translations are. Given there's other scheme stuff in `setup_validations.rb` it probably goes in there?

The postcode one above feels like it probably also doesn't live in this file, even though I think that will be applied to both sales and lettings - the best way of handling that might need a little more thinking (e.g. at the moment I think we just duplicate the UPRN validation in property validations for sales/lettings, which may be the way to go but is also not necessarily ideal).

Reply:
The same applies to the other scheme/location related validations.

I've also got question marks over `shared_validate_partner_count` and `validate_owning_organisation_data_sharing_agreement_signed` similar to the postcode one - I think it's essentially the difference between generic validations that apply to any question fitting a certain type and specific validations applied to specified questions, even though they are shared between lettings and sales.

Reply:
I've moved the location, scheme and tenancy start date validation methods to lettings setup validations. What's the downsides to leaving the postcode method in the shared validations file? I'm hesitant to duplicate the method. Since it's in the shared validations file and starts with "validate" it'll run everytime right? But that would also be the same situation even if moved, right?

Reply:
It'll run here, it's entirely about what makes sense as code structure. 

The majority of this file is generic, i.e. validations that will run on any question that matches a certain condition (e.g. any numeric question with a range set). That's not true of the postcode validation, the partner count validation, or the owning org data sharing agreement validation, which are all specific validations on specific questions. 

I think the key is where would you look for them - I think I'd look for postcode validations in `property_validations` where the uprn validation is (and that is duplicated for lettings and sales), I'd look for partner count validation in `household_validations` (and in fact in that case the validation is there and uses this as a shared function it calls with different max_people parameters), and I'd look for an owning org dsa validation in `setup_validations`. 

I think the nice way to set it up would be to have a file of generic validations and separately have e.g. shared household validations used by both lettings and sales, as well as individual files for lettings-only and sales-only things. But that would be a bit much to do right now. I think duplication between sales and lettings is annoying but not as huge a problem as not knowing where to look for things, particularly given they do typically have slight differences (e.g. which error message). (That's why I said it might need more thinking about the best solution)

Reply:
Moved those methods around as you've described.